### DISCUSSION TITLE
Does Value filter support joint value of multiple AWS resources tags

### DISCUSSION THREAD

**Discussion by zczc0128** [cloud-custodian/cloud-custodian#7416] [General]
In out company we define a group of tags to identify our AWS resource, like TagA, TagB,TagC, all our AWS resources must have these 3 tag, and tag value is pre-defined.

Now we're looking into if we can use Cloud Custodian to do tag validation for our AWS resource, from the document I know it supports check existence of specific tag key, and also support validating specific tag value based on a value list in S3 file.

But didn't see an example about validate joint value of multiple tag key, in our case other than checking existence of these 3 tag key, we also need validate their value(joint value) based on pre-defined value list

In following example joint tag value "a1,b1,c1","a2,b2,c2","a3,b3,c3" are treated as valid tag

> **Tag Key:**     TagA, TagB, TagC
> **Tag Value:**     a1,       b1,      c1
> **Tag Value:**     a2,      b2      c2
> **Tag Value:**     a3       b3,     c3

So does cloud custodian support following kind of tag validation ? Thanks!

      - type: value
        key: "tag:TagA,tag:TagB,tag:TagC"
        op: in
        value_from:
          url: <joint_tag_value_list_file>

**Comment by ajkerrigan** [comment]
Hi @zczc0128 - interesting question. I don't know of any dedicate syntax for that, though you can play with JMESPath expressions in the value filter to get something that should be useful.

Adapting your example:

```
- type: value
  key: "join(`,`, sort_by(Tags[?contains([`TagA`,`TagB`,`TagC`], Key)], &Key)[].Value)"
  op: in
  value_from:
    url: <joint_tag_value_list_file>
```

Where ```Tags[?contains([`TagA`,`TagB`,`TagC`], Key)]``` will be up to 3 key/value pairs containing only the tags we want. We sort them by key so the ordering is consistent, then mush the values together into one comma-separated string.
